app/models/tipline_newsletter.rb

```diff
@@ -0,0 +1,74 @@
+class TiplineNewsletter < ApplicationRecord
+  include RssFeedHelper
+
+  belongs_to :team
+  has_paper_trail on: [:create, :destroy], save_changes: true, ignore: [:updated_at, :created_at], versions: { class_name: 'Version' }
+
+  validates_presence_of :introduction, :team
+  validates_format_of :rss_feed_url, with: URI.regexp, allow_blank: true, allow_nil: true
+  validates_inclusion_of :number_of_articles, in: 0..3
+  validates_inclusion_of :send_every, in: ['everyday', 'sunday', 'monday', 'tuesday', 'wednesday', 'thursday', 'friday', 'saturday']
+  validates_inclusion_of :language, in: ->(newsletter) { newsletter.team&.get_languages.to_a }
+
+  after_save :reschedule_delivery
+
+  def cron_notation
+    hour = self.time.hour
+    # If an offset is being passed, it's in the new format
+    if self.timezone.match?(/\W\d\d:\d\d/)
+      timezone = self.timezone.match(/\W\d\d:\d\d/)
+    else
+      timezone = self.timezone.to_s.upcase
+    end
+    # Mapping for old-style timezones not supported by Ruby's DateTime
+    timezone = {
+      'PHT' => '+0800',
+      'CAT' => '+0200'
+    }[timezone] || timezone
+    time_set = DateTime.parse("#{hour}:00 #{timezone}")
+    time_utc = time_set.utc
+    cron_day = nil
+    if self.send_every == 'everyday'
+      cron_day = '*'
+    else
+      days = (0..6).to_a
+      day = ['sunday', 'monday', 'tuesday', 'wednesday', 'thursday', 'friday', 'saturday'].index(self.send_every)
+      day += (time_utc.strftime('%w').to_i - time_set.strftime('%w').to_i)
+      cron_day = days[day]
+    end
+    "#{time_utc.min} #{time_utc.hour} * * #{cron_day}"
+  end
+
+  def body
+    content = []
+    [:first_article, :second_article, :third_article].each do |article|
+      content << self.public_send(article)
+    end
+    content.reject{ |article| article.blank? }.join("\n\n")
+  end
+
+  def build_content(cache_hash = true)
```

Comment:
ah, I'm seeing now this is a port of a function from another class! I still think it would be great if we could get tests around it and/or make a bit more straightforward, but also understand that can be tricky in cases like this. up to you

Reply:
Yes, but good point in both fronts. Actually I did both: split that method for clarity and added more unit tests for the cases in `TiplineNewsletter` unit tests suite.